### PR TITLE
Refactor dalc into celestia-node plugin

### LINKED ISSUE #51: Get `retrieveBlock` and `checkDataAvailability` calls to `dalc` work
The `dalc` should be updated so that it can make `retrieveBlock` and `checkDataAvailability` calls by connecting to a `celestia-node` full node.

### PR DESCRIPTION
**Pull request by evan-forbes**
## Description

The dalc needs access to certain celestia-node internals. This PR is using the plugin mechanism introduced in [(unmerged) #407](https://github.com/celestiaorg/celestia-node/pull/407) to better integrate with celestia-node, and refactor the dalc's celestia-node usage to fully utilize preexisting components.

This is a short term solution. In the future we will likely use native celestia-node instead of wrapping it here. After connecting this to devnet, I can confirm that this is significantly easier to use than before. The setup is identical to a normal celestia-node setup, with the addtional `optimint-server.toml` config file added.

closes #51

### CODE REVIEW

File: server/test/server_integration_test.go [hunk 1]
```diff
@@ -3,82 +3,74 @@ package test
...
+
+// 	req := dalc.RetrieveBlockRequest{
+// 		Height: optimintBlock.Header.Height,
+// 	}
+// 	resp, err := dalcClient.RetrieveBlock(context.TODO(), &req)
+// 	require.NoError(t, err)
+// 	assert.Equal(t, dalc.StatusCode_STATUS_CODE_SUCCESS, resp.Result.Code)
+// 	assert.Equal(t, optimintBlock, resp.Block)
+// 	srv.Stop()
+// }
```
[the remark this answers is not in the document]
Comment (evan-forbes) [reply server/test/server_integration_test.go]: deleted :slightly_smiling_face:

**Comment by evan-forbes** [comment]
we are currently experiencing this bug, as we are using the latest version of celestia-node here. We will either need to change to a stable version, or update as soon as this is fixed.

https://github.com/celestiaorg/celestia-node/issues/511

**Comment by adlerjohn** [comment]
> wait or downgrade?

Depends on the timeline of fixing the mentioned issue. What's the timeline?

**Review by tzdybal — APPROVED** [review]
utACK. Looks fine.

**Comment by tzdybal** [comment]
> what do we think? wait or downgrade? Everything seems to be working, an error is just thrown

I would merge this and keep an eye on https://github.com/celestiaorg/celestia-node/issues/511.

**Review by jbowen93 — CHANGES_REQUESTED** [review]
The Makefile (build function), docker/Dockerfile, and entrypoint.sh should be updated. 

For Makefile.build you need to do  `go build ./cmd/celestia`

entrypoint.sh needs to be changed to `if [ "$1" = 'celestia' ]; then` on line 5

The Dockerfile needs to copy from `/dalc/celestia` to `/root/celestia` on line 17 and line 24 needs to be `CMD ["celestia"]`.

**Comment by jbowen93** [comment]
I'm getting the following errors when attempting to run this as a docker container
```
2022-03-17T22:18:57.385Z	INFO	node	node/init.go:30	Initializing Light Node Store over '/celestia-light'
2022-03-17T22:18:57.386Z	INFO	node	node/init.go:66	Saving config	{"path": "/celestia-light/config.toml"}
2022-03-17T22:18:57.386Z	INFO	dalc/server	server/plugin.go:29	Config already exists	{"path": "/celestia-light/optimint_server.toml"}
2022-03-17T22:18:57.386Z	INFO	node	node/init.go:91	Node Store initialized
len parsed 35 1
2022-03-17T22:18:57.432Z	INFO	badger	v2@v2.2007.4/levels.go:183	All 0 tables opened in 0s

2022-03-17T22:18:57.439Z	ERROR	node/services	services/service.go:112	initializing store failed: protocol not supported
2022-03-17T22:18:57.439Z	INFO	watchdog	go-watchdog@v1.2.0/watermarks.go:16	initialized watermark watchdog policy; watermarks: []; thresholds: [67467048960 80960458752 94453868544 114693983232 121440688128 124814040576 128187393024]
2022-03-17T22:18:57.439Z	INFO	watchdog	go-watchdog@v1.2.0/watchdog.go:471	initialized heap profile capture; threshold: 0.900000; max captures: 10; dir: /celestia-light
2022-03-17T22:18:57.439Z	INFO	header	header/p2p_server.go:37	p2p-server: listening for inbound header requests
2022-03-17T22:18:57.439Z	INFO	node	node/node.go:113	started Light Node
The p2p host is listening on:
*  /ip4/192.167.10.14/tcp/2121/p2p/12D3KooWJSus3RxUQpNWwDH8XoxrwtufttUFbNCnH2ULxdDox2bB
2022-03-17T22:18:57.439Z	ERROR	header	header/sync.go:170	getting trusted head	{"err": "header/store: no chain head"}
```
it sounds like the header/sync error is expected. Is the `ERROR	node/services	services/service.go:112	initializing store failed: protocol not supported` related?

**Comment by evan-forbes** [comment]
> Is the ERROR node/services services/service.go:112 initializing store failed: protocol not supported related?

I don't know if that's related or not, but it does occur running the normal celestia-node (latest commit main) on devnet.

While we might want to merge this version to master, I think we should definitely create a version of the plugin/dalc based on the latest stable release of celestia-node as well. That way we can feel more comfortable pointing potential devs to it, while also having a version that we can test the latest and greatest on.

**Pull request merged**